server/trillian_log_server.go

```diff
@@ -89,8 +89,53 @@ func (t *TrillianLogServer) QueueLeaves(ctx context.Context, req *trillian.Queue
 }
 
 // GetInclusionProof obtains the proof of inclusion in the tree for a leaf that has been sequenced.
+// Similar to the get proof by hash handler but one less step as we don't need to look up the index
 func (t *TrillianLogServer) GetInclusionProof(ctx context.Context, req *trillian.GetInclusionProofRequest) (*trillian.GetInclusionProofResponse, error) {
-	return nil, ErrNotImplemented
+	// Reject obviously invalid tree sizes and leaf indices
```

Comment:
Most of the duplication is in the tests. Apart from parameter validation they call the same function to retrieve the proofs.